### DISCUSSION TITLE
Xarray not masking FillValue

### DISCUSSION THREAD

**Discussion by tlsw231** [pydata/xarray#6179] [Q&A]
I am working with an ocean dataset and am plotting some field (see attachment) versus depth at some (lon, lat) location. Those weirdly large values at the bottom three vertical levels are the FillValues in the netcdf file. I am not sure why xarray is not masking these automatically for this particular dataset as in others, I would not even see those points. So I tried to manually change the FillValue attribute in the encoding but it does not seem to work:

ds['var'].encoding['_FillValue']=np.nan
ds['var'].encoding['missing_value']=np.nan

When I plot the field ds['var'], I am still seeing the FillValue in the plot. Is there some other way to modify the encoding?

[Edits: 1. I have since tried `mask_and_scale=True` in the `xr.open_dataset` step but to no avail.
2. `temp=xr.where(ds['var'] != ds['var'].encoding['_FillValue'],ds['var'],np.nan)]` also does not work. ]

![WTT](https://user-images.githubusercontent.com/90293171/150409604-c4318bf1-52b1-446c-9ca1-26c6489e67ef.png)

**Comment by dcherian** [comment]
`_FillValue` cannot be nan it must something like 1e37.

There must be some floating point mismatch between what's in `_FillValue` and the actual value. Can you check that please?

**Reply by tlsw231** [reply]
Here are the values I get from the encoding:

`'missing_value': 9.96921e+36`
` '_FillValue': 9.96921e+36`

Here is the value I get from the actual field:

`ds.WTT.isel(nlat=100,nlon=200,time=-1,z_w_top=-1).values`
`array(9.9692087e+36)`

**Reply by tlsw231** [reply]
For now, I am getting around this in a very inelegant way:

```
for var in ds.data_vars:
    if len(ds[var].shape)==4:  # This picks only 4d variables with dims (time,z_t,lat,lon)
        ds[var]=xr.where(ds[var] < 0.5e37,ds[var],np.nan)
```
